feat: allow choosing favorite buttons in bottom navigation bar

shuuji3 opened the pull request:
resolves #2757

shuuji3 commented on components/nav/NavBottom.vue:
```diff
@@ -1,12 +1,12 @@
-const { notifications } = useNotifications()
-const lastAccessedNotificationRoute = useLocalStorage(STORAGE_KEY_LAST_ACCESSED_NOTIFICATION_ROUTE, '')
-const lastAccessedExploreRoute = useLocalStorage(STORAGE_KEY_LAST_ACCESSED_EXPLORE_ROUTE, '')
+const navButtons = currentUser.value
+  ? [NavButtonHome, NavButtonSearch, NavButtonNotification, NavButtonMention]
+  : [NavButtonExplore, NavButtonLocal, NavButtonFederated]
```
These buttons will be customizable by the following PRs.

userquin commented on components/settings/SettingsBottomNav.vue:
```diff
@@ -0,0 +1,81 @@
+    <button
+      v-for="{ name, label, icon } in availableNavButtons"
+      :key="name"
+      btn-text flex="~ gap-2" p2 border="~ base rounded" bg-base ws-nowrap
```
maybe you can add `selected` option to `NavButton` to simply logic...

can you add these attributes?
```vue
      type="button"
      role="switch"
      :aria-checked="selectedNavButtonNames.includes(name)"
```

and remove `disable` attribute?

You also need to add remove logic...

userquin replied:
I'll check it next weekend on my local, don't apply changes yet

shuuji3 replied:
Thanks for suggestion for button styling! 


I'm actually still explorering what kind of UI would be better overall. A bit more adjustment would be included until finalizing. Maybe I'll add some ordering button or better remove button etc. later for the easy arrangement.

userquin replied:
Maybe an editable forn? Check push notification in notifications settings (you will needd to enable them, dont forget to remove once you've check it)

shuuji3 replied:
OK, I was able to implement the toggle button logic now. 🙂

shuuji3 commented:
Now it's ready to review.

- Each button can add/remove the button to/from the nav bar.
- Added "Clear" button - remove all buttons from nav bar
- Added "Reset" button ()

<img src="https://github.com/elk-zone/elk/assets/1425259/6c154d1b-a8c5-4379-a01e-7293e647d72c" width="400"> 

![Screenshot from 2024-04-12 17-07-30](https://github.com/elk-zone/elk/assets/1425259/ab3b538d-9ebe-4276-bfd5-16bb7822a525)

patak-dev reviewed (approved):
This is incredible! A lot of people will be very happy when this feature is released.

The current way to show what buttons were available wasn't very clear to me, and I didn't discover easily that I should first clear then add the buttons to get the order I want. I don't have a better proposal that doesn't involve complex drag and drop so I think we should merge this for now. Users aren't going to use this all the time. If we have a better idea later, we can improve it.

shuuji3 commented:
@patak-dev I just realized that in the normal use case, users want to create a different button arrangement than the current one so there is no need to show the current ones. #2803 will set the default selection empty.

Yes, the drag and drop pattern was exactly one of the candidates I was thinking about, but avoid it for now due to complexity for now. If I come up with a better UI, I'll make another PR.
